src/argilla/client/feedback.py

```
@@ -668,11 +668,17 @@ def from_argilla(
         cls.argilla_id = existing_dataset.id
         self = cls(
             fields=[
-                FieldSchema.construct(**field.dict())
+                TextField.construct(**field.dict())
+                if field.settings["type"] == "text"
+                else FieldSchema.construct(**field.dict())
                 for field in datasets_api_v1.get_fields(client=httpx_client, id=existing_dataset.id).parsed
             ],
             questions=[
-                QuestionSchema.construct(**question.dict())
+                RatingQuestion.construct(**question.dict())
+                if question.settings["type"] == "rating"
+                else TextQuestion.construct(**question.dict())
+                if question.settings["type"] == "text"
+                else QuestionSchema.construct(**question.dict())
```

I think this line should raise an error instead of parsing an unexpected question type. What do you think @alvarobartt?

---

We can either raise and exception, or assume that the Argilla Server may have one version e.g. 1.9.0 with support for a new question type, while the Argilla Python Client is another version e.g. 1.8.0 with support for that question type, so raising and error may not be desired in that scenario.

Anyway, if you think we're safer raising the exception on `ValidationError`, then I'll change that 👍🏻 LMKWYT!

---

I see. You mean allowing Python clients to provide question types without version upgrades, right? If we have no specific validation, could make sense. Anyway, also it makes sense to me also force a version upgrade to support new types.

---

Sure, then I'll just allow the current types and ignore anything that is not explicitly supported in the Argilla version used in the Python client 👍🏻